controllers/configuration_controller.go

```diff
@@ -89,16 +95,33 @@ func (r *ConfigurationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, erro
 	klog.InfoS("reconciling Terraform Configuration...", "NamespacedName", req.NamespacedName)
 	applyJobName := configurationName + "-" + string(TerraformApply)
 
-	// Terraform destroy
 	if err := r.Get(ctx, req.NamespacedName, &configuration); err != nil {
-		klog.InfoS("performing Terraform Destroy", "Namespace", req.Namespace, "Name", req.Name)
 		if kerrors.IsNotFound(err) {
+			klog.ErrorS(err, "unable to fetch Configuration", "NamespacedName", req.NamespacedName)
+			err = nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	if configuration.ObjectMeta.DeletionTimestamp.IsZero() {
+		if !meta.FinalizerExists(&configuration, configurationFinalizer) {
+			meta.AddFinalizer(&configuration, configurationFinalizer)
```

Comment:
It seems that we seldom add a finalizer during deletion. How about adding it in the very begining?

Reply:
If we don't use Finalizer, when we delete Configuration, it will be deleted directly, but at this point, the state of the Configuration and the state of the cloud resource will not be consistent.

Reply:
You can add the finalizer at the very beginning.

Reply:
Sorry, I'm not very proficient in Using Finalizer.
therefore, I am not sure about your mean. 
Can you given me an example? Let me know, what I should to do
thanks.